internal/handler/checkin/checkin.go

```diff
@@ -4,26 +4,41 @@ import (
 	"github.com/RobinHoodArmyHQ/robin-api/internal/env"
 	"github.com/RobinHoodArmyHQ/robin-api/internal/handler/contract"
 	"github.com/RobinHoodArmyHQ/robin-api/internal/repositories/checkin"
+	"github.com/RobinHoodArmyHQ/robin-api/internal/repositories/event"
 	"github.com/RobinHoodArmyHQ/robin-api/pkg/nanoid"
 	"github.com/gin-gonic/gin"
 	"net/http"
 )
 
 func CreateCheckInHandler(c *gin.Context) {
-	checkIn := &checkin.CreateCheckInRequest{}
-	err := c.BindJSON(checkIn)
+	req := &checkin.CreateCheckInRequest{}
```

Comment:
This is `internal/repositories/checkin/checkin.go` model which we're taking as api input and directly storing in db. Ideally we should define separate struct for api input in `models` package -> add necessary validations (like length of photos) -> transform to repository checking struct -> store in db